src/util/diproche/diprocheInterface.test.ts

```diff
@@ -0,0 +1,89 @@
+import { emptyIssueList, listAllIssues} from "../../issueHandling/issueMapping";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+import {addPredicate, getVocabErrors, Mode} from "./diprocheInterface";
+
+describe("getErrors", () => {
+	it("Returns all errors present", () => {
+	 const userInput = "Angenommen A ist falshc. Dann ist A => B wahr.";
+	 getVocabErrors(userInput);
+	 expect(listAllIssues()).toEqual(getAllIssues(userInput));
+	});
+});
+
+describe("addPredicate", () => {
+	describe("Normal cases", () => {
+		describe("Adding the predicates works for a normal string", () => {
+			it("Adds the prop-logic predicate correctly", () => {
+				const pred = "propositionalLogic";
+				expect(addPredicate("hallo", pred)).toEqual("diproche(hallo).");
+			});
+			it("Adds FO-logic predicate correctly", () => {
+				const pred = "firstOrderPredicateLogic";
+				expect(addPredicate("hallo", pred)).toEqual("diproche_fo(hallo).");
+			});
+			it("Adds test-predicate correctly", () => {
+				const pred = "test";
+				expect(addPredicate("hallo", pred)).toEqual("teste(hallo).");
+			});
+		});
+	});
+	describe("Edge cases", () => {
+		describe("Empty string as an input", () => {
+			it("Adds the prop-logic predicate to an empty string", () => {
+				const pred = "propositionalLogic";
+				expect(addPredicate("", pred)).toEqual("diproche().");
+			});
+			it("Adds FO-logic predicate to an empty string", () => {
+				const pred = "firstOrderPredicateLogic";
+				expect(addPredicate("", pred)).toEqual("diproche_fo().");
+			});
+			it("Adds test-predicate to an empty string", () => {
+				const pred = "test";
+				expect(addPredicate("", pred)).toEqual("teste().");
+			});
+		});
+	});
+	describe("Error cases", () => {
+		describe("Predicate is invalid", () => {
```

Comment:
In my opinion you are nesting describes too much. This makes it harer to read and in this case you put two extra describes for one test. I think normal cases, errors cases and limit cases don't have to be explicitely mentioned and can be described in the test name like "Will throw a SomeErrorException when the predciate is invalid".

Reply:
I wouldn't go as far saying one style is superior to the other, but in my mind this cascade of describes results in more structured testcases in general. I can see why it is redundant from your point of view, because it just adds more text to read while providing no additional functionality or testcoverage. However, in the case I add more and more tests later (which most likely will happen), one might find the repeated partition useful for understanding the testcases better.

Mostly it's just a choice of design, but for the time being, I prefer the nested describes. But I can see where you're coming from.

Reply:
When using your enum-based Ansatz, this particular error case got redundant anyways, since the linter doesn't allow for an invalid Predicate anyways. However, the topic of this choice of design remains.